lib/engines/hcktest/tests.rb

```diff
@@ -285,8 +285,21 @@ def delete_old_remote(test_name)
       @project.result_uploader.delete_file(r_name)
     end
 
+    def last_queued_test_running?
+      results = @tools.list_test_results(@queued_test['id'], @target['key'], @client.name, @tag)
+      last_result = results.max_by { |k| k['instanceid'].to_i }
+
+      last_result['status'] == 'Running' || last_result['status'] == 'InQueue'
+    end
+
     def all_tests_finished?
-      status_count('InQueue').zero? && current_test.nil?
+      # After the first Pass, HLK always reports a Pass as
+      # a status for this test even if it was queued again.
+      # Check that the result of the last queued test is not
+      # indicated as running.
+      # current_test can have a race condition when AutoHCK
+      # requests it earlier than the HLK update test execution state.
+      status_count('InQueue').zero? && current_test.nil? && !last_queued_test_running?
     end
```

Comment:
Shouldn't `current_test` be fixed instead of fixing `all_tests_finished?` as `current_test` is also used in other places?